### ISSUE TITLE
Possible error in PLE0605

### ISSUE THREAD

**Issue by nstarman** [astral-sh/ruff#7672]
In https://github.com/astropy/astropy/pull/15385 we are trying to fix some ``__all__`` statements and ran into the following

```python
__all__ = [...]
logging_levels = ["NOTSET", ...]
...
__all__ += logging_levels
```

My suggested fix was 

```python
__all__ = [...]
__all__ += (logging_levels := ["NOTSET", ...])
```

However PLE0605 is still complaining. I don't think this is correct since the above should be, from the "perspective" of `__all__` equivalent to 

```python
__all__ = [...]
__all__ += ["NOTSET", ...]
```

which passes the check.

The inclusion of the walrus operator for assigning to another variable makes the AST more complicated but should be equivalent statically to excluding the walrus.

**Comment by zanieb** [comment]
Thanks for the report!

Hm... in the example:

```python
from first import a, b

__all__ = ["a"]
__all__ += (foo := ["b"])
```

pyright reports the symbol `b` as unused and displays the warning `Operation on "__all__" is not supported, so exported symbol list may be incorrect (reportUnsupportedDunderAll)`.

Although I agree this is syntactically equivalent, if it's not supported by type checkers I'm not sure we should treat it as valid.

**Comment by charliermarsh** [comment]
I think it's reasonable to support if the code changes aren't too tricky :)
